### PR TITLE
added support for multiple-compose files

### LINKED ISSUE #275: Support multiple compose files
In my environment I have multiple docker-compose files: one "base", and then some more to enable additional services/options. These additional docker-compose files override/modify the base image settings.

It would be nice if kompose supported this natively: `kompose -f base.yml -f mod.yml -f other-mod.yml convert`.

Currently it simply ignores all but the last-mentioned file.

### PR DESCRIPTION
**Pull request by procrypt**
Fix #275 

```bash
$ kompose -f docker-compose.yml,docker-guestbook.yml convert
INFO[0000] file "redis-master-service.json" created     
INFO[0000] file "redis-slave-service.json" created      
INFO[0000] file "redis-service.json" created            
INFO[0000] file "web-service.json" created              
INFO[0000] file "frontend-service.json" created         
INFO[0000] file "redis-master-deployment.json" created  
INFO[0000] file "redis-slave-deployment.json" created   
INFO[0000] file "redis-deployment.json" created         
INFO[0000] file "web-deployment.json" created           
INFO[0000] file "frontend-deployment.json" created 
```
cc @kadel @surajssd

### CODE REVIEW

**Comment by kadel** [comment]
I'm not sure if comma separated list is right here :-(
With docker-compose it looks like this - `docker-compose -f docker-compose.yml -f docker-guestbook.yml up`
I think that we should do it in the same way as docker-compose does it.

@cdrage how would this work with https://github.com/kubernetes-incubator/kompose/pull/304 ?
Comma separated or multiple -f arguments?

**Comment by cdrage** [comment]
@kadel would be easy to implement via #304 once it's merged. But yes, multiple -f arguments would work with it.

**Comment by kadel** [comment]
Lets wait for #304. Thank we can revisit this.

**Comment by surajssd** [comment]
@procrypt unit tests, docs needed as well! 

Also I thought we are trying to implement this the docker-compose cli way which is 

```
docker-compose -f foo.yml -f bar.yml up
```
as opposed to

```
docker-compose -f foo.yml,bar.yml up
```

File: script/test/cmd/tests.sh [hunk 1]
```diff
@@ -103,6 +103,7 @@ convert::expect_success "kompose --bundle $KOMPOSE_ROOT/script/test/fixtures/bun
 convert::expect_success_and_warning "kompose --bundle $KOMPOSE_ROOT/script/test/fixtures/bundles/dsb/docker-voting-bundle.dsb convert --stdout" "$KOMPOSE_ROOT/script/test/fixtures/bundles/dsb/output-k8s.json" "Service cannot be created because of missing port."
 
 ######
+<<<<<<< 73418310ac1e607b41d0af2588ed7a808a3c85fa
```
Comment (surajssd) [inline script/test/cmd/tests.sh]: remove this line.

File: script/test/fixtures/multiple-compose-files/docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,10 @@
+web:
+  image: tuna/docker-counter23
+  ports:
+    - "5000:5000"
+  links:
+    - redis
+redis:
+  image: redis:3.0
+  ports:
+    - "6379"
```
Comment (surajssd) [inline script/test/fixtures/multiple-compose-files/docker-compose.yml]: a real world example where things get over-written will make more sense as opposed to mixing two apps.

Comment (surajssd) [inline script/test/fixtures/multiple-compose-files/docker-compose.yml]: things as in same parameters like env vars from 1st docker-compose file get over-written in second!

File: script/test/cmd/tests.sh (hunk 1 above)
Comment (procrypt) [inline script/test/cmd/tests.sh]: I pushed this by mistake.

**Comment by procrypt** [comment]
@surajssd @containscafeine I have updated the PR. Please review
```bash
$ kompose -f docker-compose.yml -f docker-guestbook.yml convert 
INFO[0000] file "frontend-service.json" created         
INFO[0000] file "mlbparks-service.json" created         
INFO[0000] file "mongodb-service.json" created          
INFO[0000] file "redis-master-service.json" created     
INFO[0000] file "redis-slave-service.json" created      
INFO[0000] file "frontend-deployment.json" created      
INFO[0000] file "mlbparks-deployment.json" created      
INFO[0000] file "mongodb-deployment.json" created       
INFO[0000] file "mongodb-claim0-persistentvolumeclaim.json" created 
INFO[0000] file "redis-master-deployment.json" created  
INFO[0000] file "redis-slave-deployment.json" created 
```
```bash
$ kompose -f docker-compose.yml -f docker-guestbook.yml convert 
INFO[0000] file "mlbparks-service.json" created         
INFO[0000] file "frontend-service.json" created         
INFO[0000] file "mongodb-service.json" created          
INFO[0000] file "redis-master-service.json" created     
INFO[0000] file "redis-slave-service.json" created      
INFO[0000] file "frontend-deployment.json" created      
INFO[0000] file "mlbparks-deployment.json" created      
INFO[0000] file "mongodb-deployment.json" created       
INFO[0000] file "mongodb-claim0-persistentvolumeclaim.json" created 
INFO[0000] file "redis-master-deployment.json" created  
INFO[0000] file "redis-slave-deployment.json" created 
```
The test are failing for this particular feature because the artifacts created are not in any particular order. When we run the `kompose convert` command sometimes `mlbparks-service.json` will be crated first and sometimes `frontend-service.json`and this makes it difficult to match the output in the tests for this feature. We need to think of a different testing strategy to test this rather than matching the output.

**Comment by surajssd** [comment]
@procrypt even if the order is upside down, it should not cause any problems, because that is what happens for other conversion also, the order may not be same and still `jq` does the matching.

**Comment by surajssd** [comment]
@procrypt one last bit, add info about this awesome fetaure to docs! Rest LGTM

**Comment by procrypt** [comment]
@surajssd Thanks I'll update the docs.

**Comment by surajssd** [comment]
@procrypt thanks for awesome work, code + tests + docs :+1: :tada:

**Comment by procrypt** [comment]
@surajssd Thank you all your help 😄

**Pull request merged**